lib/redis_memo/memoize_method.rb

```diff
@@ -81,14 +82,40 @@ def self.extract_dependencies(ref, *method_args, &depends_on)
 
   def self.get_or_extract_dependencies(ref, *method_args, &depends_on)
     if RedisMemo::Cache.local_dependency_cache
-      RedisMemo::Cache.local_dependency_cache[ref] ||= {}
-      RedisMemo::Cache.local_dependency_cache[ref][depends_on] ||= {}
-      RedisMemo::Cache.local_dependency_cache[ref][depends_on][method_args] ||= extract_dependencies(ref, *method_args, &depends_on)
+      RedisMemo::Cache.local_dependency_cache[ref.class] ||= {}
```

Comment:
> Before we were passing in self as ref where self is a specific instance of RedisMemo::MemoizeMethod.

Self is a specific instance of the object on which the method is memoized.

> Instead we can just grab the class via ref.class because dependencies will be the same for any member of the class. 

This is the general idea - that dependencies might be the same for any member of the class (though this is not always true). A dependency can depend on a specific attribute of the instance itself, but the instance is already passed in as part of the parameters (when we call `depends_on.parameters` below, the first argument is always a reference to the instance.)